wgpu-types/src/lib.rs

```diff
@@ -1022,17 +1372,25 @@ pub struct Extent3d {
     pub depth: u32,
 }
 
+/// Describes a [`Texture`].
 #[repr(C)]
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub struct TextureDescriptor<L> {
+    /// Debug label of the texture. This will show up in graphics debuggers for easy identification.
     pub label: L,
+    /// Size of the texture. For a 1D/2D texture, the unused sizes will be 1.
     pub size: Extent3d,
+    /// Mip count of texture. For a texture with no extra mips, this must be 1.
     pub mip_level_count: u32,
+    /// Sample count of texture. If this is not 1, texture must be read from shaders with `texture1DMS`, `texture2DMS`, or `texture3DMS`.
```

Comment:
I think, instead of linking to the shaders from here, we should do it through the BGL (bind group layout):
  - here, specify what `BindingType` the views of this texture can be used as
  - in the `BindingType`, specify what the shader expectations are